### ISSUE TITLE
Integrating gulp for development of angular-jsdoc sources

### ISSUE THREAD

**Issue by reloaded** [allenhwkim/angular-jsdoc#76]
It would be great if angular-jsdoc can integrate its own Gulpfile.

The Gulpfile should at minimum provide the following tasks.

- analysis: Run JSHint and JSCS (JS Code Style) and fail if either of those throw warnings or errors.
- docs: Delete the angular-template/docs and default/docs and regenerate them from sample-codes.

**Comment by reloaded** [comment]
Whoops! I just noticed that there is already a Gulpfile present! It doesn't have any helpful developr tasks other than incrementing the version number.

Would you be open to a PR that adds additional tasks and gulp plugin integrations? Ie JSHint and JSCS?

**Comment by allenhwkim** [comment]
I intentionally did not add a gulp file for simplcity since npm command can handle it all. npm has very extensive support for these command. e.g., npm start, npm publish, npm test, etc. I found this article useful. http://blog.keithcirkel.co.uk/how-to-use-npm-as-a-build-tool.

TLDR; getting away from grunt/gulp, and getting back into npm.
